Sources/SwiftSyntax/Syntax.swift

```diff
@@ -382,7 +382,10 @@ struct SyntaxData: Sendable {
 /// `SyntaxDataArena` manages the entire data of a `Syntax` tree.
 final class SyntaxDataArena: @unchecked Sendable {
   /// Mutex for locking the data when populating layout buffers.
-  private let mutex: PlatformMutex
+  ///
+  /// - Note: `UnsafeMutableRawPointer` + casting accessor is a workaround to silence the warning 'cannot bypass resilience'.
+  private let _mutex: UnsafeMutableRawPointer!
```

Comment:
Why does this nave to be an IUO? Shouldn’t we be able to make this a `UnsafeMutableRawPointer`?

Reply:
We have a platforms it is actually `nil`. I used `!` because the `PlatformMutex.opaque` type is `UnsafeMutableRawPointer!` (by default by ClangImporter)

Reply:
Oh, I see. Should we just make it a proper optional in that case?

Reply:
Okay, I don't see any drawbacks for using optional. Done